src/main.rs

```diff
@@ -50,26 +51,20 @@ impl EventHandler for Handler {
             return;
         }
 
-        let re: Regex = Regex::new(r"^/?(r/).+").expect("Failed to initialise regex pattern");
-
-        if !re.is_match(&message.content) {
-            return;
-        }
+        let path = match content_to_path(&message.content) {
+            Some(path) => path,
+            None => return,
+        };
 
         message
             .delete(ctx.http.clone())
             .await
             .expect("Couldn't delete message");
 
-        let split_message: Vec<&str> = message.content.split('/').collect();
-
         let msg = message
             .channel_id
             .send_message(&ctx.http, |m| {
-                m.content(format!(
-                    "{} https://reddit.com/r/{}",
-                    message.author, split_message[1]
```

Comment:
which if you then index in here would return not what you're expecting.

Also, it's generally better to use `.get(1).expect("...")` rather than `[1]`, because you have to explain why what you're doing can't panic (`[x]` slicing notation will panic if it's out of range)